app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt

```diff
@@ -603,24 +604,20 @@ class SummaryProductFragment : BaseFragment(), ISummaryProductPresenter.View {
         withContext(Main) {
             when (state) {
                 is ProductInfoState.Data -> {
-                    val analysisTags = state.data
-
-                    binding.analysisContainer.visibility = View.VISIBLE
-
+                    binding.analysisContainer.isVisible = true
                     binding.analysisTags.adapter = IngredientAnalysisTagsAdapter(
                         requireContext(),
-                        analysisTags,
+                        state.data,
                         picasso,
                         sharedPreferences
                     ).apply adapter@{
                         setOnItemClickListener { view, _ ->
-                            IngredientsWithTagDialogFragment.newInstance(
+                            val fragment = IngredientsWithTagDialogFragment.newInstance(
                                 product,
                                 view.getTag(R.id.analysis_tag_config) as AnalysisTagConfig
-                            ).run {
-                                onDismissListener = { filterVisibleTags() }
-                                show(childFragmentManager, "fragment_ingredients_with_tag")
-                            }
+                            )
```

Comment:
What changes? Could you explain with a comment so that we do not repeat the same mistake?

Reply:
sure!
the tricky part here is in how `run` works - it creates a lambda and passes the fragment as a parameter inside, other words it creates a scoped place with this fragment only
so the code looks similar w/o `run` but works differently, the fragment has a complex lifecycle and inside `run` lambda it doesn't pass through all method and doesn't attach to the child fragment manager (because lambda has own scope)
so when you call `show` fragment manager cant show it

in a result inside `run` we can setup a listener and it will work, but showing doesn't work